nomenclature/testing.py

```diff
@@ -63,35 +68,36 @@ def _check_mappings(
         raise FileNotFoundError(f"Mappings directory not found: {path / mappings}")
 
 
-def _collect_RequiredData_errors(
-    required_data_dir: Path, dsd: DataStructureDefinition
+def _collect_processor_errors(
+    path: Path, processor: Processor, dsd: DataStructureDefinition
```

Comment:
I think the type hint for processor is not correct, the way it's written right now, it would expect an instance of `Processor` not the class itself.
Since `Processor` currently only implements `apply` and not `from_file` or `validate_with_definition` we'd have to use the following:

```suggestion
    path: Path, processor: type[RequiredDataValidator] | type[DataValidator], dsd: DataStructureDefinition
```

I thought about adding `from_file` and `validate_with_definition` to `Processor` as abstract methods but `RegionProcessor` also inherits from `Processor` and implements neither function. 
I think the above is the best fix.

Reply:
Thanks, done